engine/runtime/src/main/java/org/enso/interpreter/runtime/Context.java

```diff
@@ -463,4 +468,31 @@ public TruffleLogger getLogger(Class<?> klass) {
   public long clockTick() {
     return clock.getAndIncrement();
   }
+
+  /** Return the {@code Standard.Base.Data.Time.Date} constructor.
+   *
+   * @return optional with {@link AtomConstructor} for the date, if it can be found
+   */
+  public Optional<AtomConstructor> getDateConstructor() {
+    if (date == null) {
+      CompilerDirectives.transferToInterpreterAndInvalidate();
+      ensureModuleIsLoaded("Standard.Base.Data.Time.Date");
+      Optional<Module> dateModule = findModule("Standard.Base.Data.Time.Date");
+      if (dateModule.isPresent()) {
```

Comment:
why not `dateModule.ifPresent(module -> ...)`. is that even possible that it is not loaded if `getDateConstructor` is invoked?

Reply:
It could happen if the Enso `Language` is manipulated from `org.graalvm.polyglot.Context` API from Java - find a `Value` representing Enso function and `value.execute(LocalDate.now())` - then the conversion can call `getDateConstructor()` without `Standard.Base.Data.Time.Date` being loaded. However I don't know the details of Enso module loading yet.